src/gui/properties/propertieswidget.cpp

```diff
@@ -311,8 +311,11 @@ void PropertiesWidget::loadTorrentInfos(BitTorrent::TorrentHandle *const torrent
       label_created_by_val->setText(m_torrent->creator());
 
       // List files in torrent
-      PropListModel->model()->setupModelData(m_torrent->info());
-      filesList->setExpanded(PropListModel->index(0, 0), true);
+      BitTorrent::TorrentInfo info = m_torrent->info();
+      libtorrent::file_storage files = info.files();
+      PropListModel->model()->setupModelData(info);
+      if (!(info.filesCount() > 1 && files.name().empty()))
```

Comment:
I think you haven't understand completely what is going on here. You might need to run my patch to see the visual represenation.
In the contentview if there is subfolder it is represented by a single "root item". Which is expanded by default here.
In case the user has chosen to not create the subfolder, the view doesn't have this "root item". So if the torrent contains a bunch of subfolders the first one is expanded, which IMO isn't correct.
Sorry, I can't describe the visual elements better unless you see them.